<!-- THIS IS AN AUTOGENERATED FILE: Don't edit it directly, instead change the schema definition in the code itself. -->

_Autogenerated Tag Syntax Documentation:_

---
Cleans an alignment, such that all amino acids are measured in the same chemical environment

```xml
<AlignmentGapInserter name="(&string;)" scorefxn="(&string;)"
        max_score_diffs="(&real_cslist;)" alignment_file="(null &string;)"
        only_clean_seq_num="(1 &positive_integer;)"
        cleaned_alignment_file="(null &string;)"
        available_AAs_file="(null &string;)" nbr_e_threshold="(0.1 &real;)"
        loop_seqid_thresholds="(0.65 &real_cslist;)"
        indel_motif_radius="(2 &non_negative_integer;)"
        confidence="(1.0 &real;)" />
```

-   **scorefxn**: Name of score function to use
-   **max_score_diffs**: Comma-separated list of exclude site thresholds
-   **alignment_file**: alignment_file to be cleaned
-   **only_clean_seq_num**: Sequence number in the alignment to filter. If unspecifed all will be filtered
-   **cleaned_alignment_file**: output file
-   **available_AAs_file**: input file
-   **nbr_e_threshold**: res res energy for neighbor classification
-   **loop_seqid_thresholds**: Sequence identity thresholds of indel motif matching loops
-   **indel_motif_radius**: If indel motif radius is 2 then 2 residues will be included up and downstream from target position
-   **confidence**: Probability that the pose will be filtered out if it does not pass this Filter

---
